jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/oracle/OracleSqlGenerationVisitor.java

```diff
@@ -263,10 +264,13 @@ private String getProjectionString(final SqlColumn column, final String projecti
         final AbstractSqlDialect dialect = (AbstractSqlDialect) getDialect();
         final String typeName = ColumnAdapterNotes
                 .deserialize(column.getMetadata().getAdapterNotes(), column.getMetadata().getName()).getTypeName();
-        if ((typeName.startsWith("TIMESTAMP") && (((OracleSqlDialect) dialect).getImportType() == ImportType.JDBC))
-                || typeName.startsWith("INTERVAL") || typeName.equals("BINARY_FLOAT")
-                || typeName.equals("BINARY_DOUBLE") || typeName.equals("CLOB") || typeName.equals("NCLOB")) {
+        if (typeName.startsWith("INTERVAL") || typeName.equals("BINARY_FLOAT") || typeName.equals("BINARY_DOUBLE")
+                || typeName.equals("CLOB") || typeName.equals("NCLOB")) {
             return "TO_CHAR(" + projectionString + ")";
+        } else if (typeName.startsWith("TIMESTAMP")
```

Comment:
Using `startsWith` has the potential to cause trouble later in the event of a newly introduced (and incompatible) timestamp format on the Oracle side.

It is safer to check for the explicit names.

Reply:
Changed to type numbers

Reply:
Okay.